### PR TITLE
feat(pt): add group_property training for labels defined over multiple frames

### PR DESCRIPTION
**Pull request by zirenjin**
## Summary

A property label sometimes belongs to a set of structures rather than to one frame.
An OER catalyst overpotential is determined jointly by the clean, `O*`, `OH*` and
`OOH*` slabs. A polymer property is determined by the repeat unit together with the
end groups, and for long chains embedding the whole chain as one system runs out of
memory, so it has to be represented by its components.

This PR adds a `group_property` training path. Per-frame representations are pooled
into one group-level embedding and the loss is computed once per group against the
single group label. Groups are declared in the data, so no new training entry point
is required.

## Data layout

Grouped data is ordinary `deepmd/npy` with three extra arrays per `set.*`:

| File            | Shape               | Meaning                                                                                     |
| --------------- | ------------------- | ------------------------------------------------------------------------------------------- |
| `group_id.npy`  | `(nframes,)`        | Group index for each frame. Frames sharing an id are pooled into one group-level prediction. |
| `weight.npy`    | `(nframes,)`        | Frame/component weight used during group pooling.                                            |
| `pool_mask.npy` | `(nframes, natoms)` | Atom mask for descriptor pooling. Values > 0 are included, values 0 are excluded.             |

Labels stay standard per-frame arrays; rows within one group must be identical because
loss is computed once per group. `fparam.npy` carries per-group side features copied to
each frame of the group.

Grouped mode is enabled when every training and validation `set.*` carries all three
markers. Partial marker sets are rejected so grouped and ungrouped systems cannot be
mixed silently.

## Core changes (`deepmd/pt`)

- `GroupPropertyFittingNet`, `GroupPropertyModel` and `GroupPropertyLoss`, registered
  alongside the existing property task.
- `group_reduce="mean" | "sum"` selects weight-normalized averaging or weighted sum for
  extensive group properties.
- `utils/grouped.py`: masked pooling helpers. Masking is applied before reduction so
  padded and virtual atoms cannot introduce NaNs.
- `utils/dataloader.py`: group-complete batching. All frames of a group land in the same
  batch, and under DDP each group is assigned to a single rank.
- `utils/argcheck.py`: schema for the new fitting type, including `group_reduce`.

## DPA-ADAPT

- `dpa_adapt/grouped/`: grouped data conversion. `Assembly` writes component arrays that
  are already in memory, plus an offline grouped dataset and a polymer builder that
  assembles repeat units and end groups into groups.
- `mark_groups()` adds markers to existing `deepmd/npy` systems in place, exposed as
  `dpa-adapt data mark-groups`. `pool_mask` is derived from `real_atom_types >= 0` when
  available.
- `Regularizer`: extra training-time loss on the downstream training path, shared by
  ordinary and grouped property training. Distinct from MFT, which regularizes the shared
  descriptor through an external auxiliary dataset and head.
- `Calibrator` and `model.calibrate(...)`: post-training prediction correction, fitted
  after training and not part of the backward pass. Features can combine the raw
  prediction, `fparam` and group statistics.
- `frozen_head`, `finetune` and `mft` all accept grouped data through the existing
  `DPAFineTuner` API.

## Tests

19 test files covering grouped conversion and writers, grouped-mode detection, pooling
semantics, the fitting net and loss, argcheck validation, DDP-safe batching, MFT with a
grouped downstream task, regularizer and calibrator, and end-to-end grouped training.

## Summary by CodeRabbit

* **New Features**
  * Added grouped-property training, batching, prediction, and evaluation.
  * Added grouped dataset creation tools, including polymer dataset generation and group-marker conversion.
  * Added post-training prediction calibration and regularizer configuration.
  * Added grouped fine-tuning and multi-task workflow support.
* **Documentation**
  * Documented grouped dataset formats, marker generation, training, calibration, and related APIs.
* **Bug Fixes**
  * Improved handling of grouped data, padding, masks, labels, caching, and distributed batching.

### CODE REVIEW

File: deepmd/pt/model/task/group_property.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+            "task_dim": self.task_dim,
+            "numb_fparam": self.numb_fparam,
+            "fparam_neuron": self.fparam_neuron,
+            "dim_case_embd": self.dim_case_embd,
+            "group_reduce": self.group_reduce,
+            "neuron": self.neuron,
+            "activation_function": self.activation_function,
+            "precision": self.precision,
+            "type_map": self.type_map,
+        }
```
[the remark this answers is not in the document]
Comment (zirenjin) [reply deepmd/pt/model/task/group_property.py:300]: I fixed `GroupPropertyFittingNet` to use a weight-bearing serialization path instead of the previous config-only payload.

The fitting payload now includes `@variables` from `state_dict()`, covering `network`, `fparam_network`, and the `case_embd` buffer, plus the relevant constructor metadata including `trainable` and `seed`. I also added `GroupPropertyFittingNet.deserialize()`, so `BaseFitting.deserialize()` now dispatches correctly and restores the state with `load_state_dict(strict=True)`.

I added regressions for both levels:
- fitting-level round trip checks parameters, `requires_grad`, `case_embd`, and predictions
- full `GroupPropertyModel.deserialize(model.serialize())` round trip checks predictions before/after

Validated with:
`pytest -s -q source/tests/pt/test_group_property.py source/tests/pt/test_group_property_fitting_net.py source/tests/pt/test_group_property_hardening.py`
→ `38 passed, 1 skipped`
